### PR TITLE
refactor BSD LB to files per request

### PR DESCRIPTION
**Pull request by nicHoch**
change the split up of packets in LB data files from by day to by "requestid-psc"

### CODE REVIEW

File: stixcore/products/levelb/binary.py [hunk 1]
```diff
@@ -125,32 +125,56 @@ def __add__(self, other):
         return type(self)(service_type=self.service_type, service_subtype=self.service_subtype,
                           ssid=self.ssid, control=control, data=data)
 
-    def to_days(self):
+    def to_files(self):
         """Split the data into a sequence of LevelB objects each containing 1 daysworth\
-        of observations.
+        of observations for contiiouse data or by request id in case of BSD.
```
Comment (samaloney) [inline stixcore/products/levelb/binary.py:130]:
```suggestion
        of observations for continuous data or by request id in case of BSD.
```

**Review by samaloney — APPROVED** [review]
Looks good

File: stixcore/products/levelb/binary.py [hunk 2]
```diff
@@ -283,6 +252,73 @@ def from_tm(cls, tmfile):
...
+                complete.append(self[seq])
+            elif (flags[seq[0]] == SequenceFlag.FIRST
+                  and flags[seq[-1]] == SequenceFlag.LAST
+                  and (((self.control['sequence_count'][seq[-1]]
+                         + (  # possible role over off 14bit psc value
+                            0 if self.control['sequence_count'][seq[-1]] >
+                            self.control['sequence_count'][seq[0]] else
+                            max(self.control['sequence_count'][seq]))) -
+                        self.control['sequence_count'][seq[0]] + 1) >= len(seq) - 1)
+                    # -1 allow one gap in the sequence
```
Comment (samaloney) [inline stixcore/products/levelb/binary.py:310]: Is this defined in the FSW, I mean could the gap be larger or was this just based on what was seen in the TM?

File: stixcore/products/levelb/binary.py [hunk 3]
```diff
@@ -125,32 +125,56 @@ def __add__(self, other):
...
+                control['index'] = control['index'] - min_index
+                data = self.data[i]
+                data['control_index'] = data['control_index'] - min_index
+
+                yield type(self)(service_type=self.service_type,
+                                 service_subtype=self.service_subtype,
+                                 ssid=self.ssid, control=control, data=data)
+        else:
+            rid = []
+            psc = []
```
Comment (samaloney) [inline stixcore/products/levelb/binary.py:158]: I take it rid is request_id  but what is psc? Maybe slight more verbose variable names would be good

File: stixcore/products/product.py [hunk 4]
```diff
@@ -582,6 +582,13 @@ def split_to_files(self):
     @classmethod
     def getLeveL0Packets(cls, levelb):
         packets = [Packet(d) for d in levelb.data['data']]
+        # packets = []
+        # for i, d in enumerate(levelb.data['data']):
+        #    try:
+        #        packets.append(Packet(d))
```
Comment (samaloney) [inline stixcore/products/product.py:588]: Delete if not used?

File: stixcore/util/scripts/idb_history.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+from stixcore.time import SCETime
+
+if __name__ == '__main__':
+    idbs = {}
+    idbs[167] = "2.26.31"
+    idbs[174] = "2.26.32"
+    idbs[178] = "2.26.33"
+    idbs[179] = "2.26.34"
+    idbs[181] = "2.26.35"
+    idbs[183] = "2.26.36"
```
Comment (samaloney) [inline stixcore/util/scripts/idb_history.py:32]:
```suggestion
    idbs = {
        167:  "2.26.31",
        174: "2.26.32",
        178: "2.26.33",
        179: "2.26.34",
        181: "2.26.35",
        183: "2.26.36",
```

File: stixcore/util/scripts/idb_history.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    data = np.column_stack((c, f, np.hstack(ver), np.hstack(idpu)))
+
+    table = QTable(data, names=['coarse', 'fine', 'version', 'idpu'])
+
+    table = table[table['idpu'] == 0]
+    table.sort('coarse')
+
+    change_indices = np.argwhere(table['version'][1:] != table['version'][:-1])
+
+    plt.clf()
```
Comment (samaloney) [inline stixcore/util/scripts/idb_history.py:71]: comment out as plots mayn't work if no GUI

**Review by samaloney — COMMENTED** [review]
Nice, few tiny comments but nothing major.

File: stixcore/products/levelb/binary.py (hunk 2 above)
Comment (nicHoch) [reply stixcore/products/levelb/binary.py:310]: yes you are right this is just based on "observation of our data": never seen something else SO FARE.
i have no better idea at the moment

File: stixcore/products/levelb/binary.py (hunk 1 above)
Comment (nicHoch) [reply stixcore/products/levelb/binary.py]: done

**Pull request merged**